src/store/slices/basketSlice/basketSlice.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  createAsyncThunk,
+  createSlice,
+  isFulfilled,
+  isPending,
+  isRejected,
+} from '@reduxjs/toolkit';
+import { QuerySnapshot } from 'firebase/firestore';
+
+import {
+  clearSnapshotCollection,
+  deleteDocSubCollectionById,
+  setDocSubCollectionById,
+  updateDocSubCollectionById,
+} from 'api';
+import { NamesDBCollection } from 'constant';
+import { handleError } from 'helpers';
+import { setErrorMessage } from 'store';
+import { IUniversalObjectArguments, ThunkApiConfig } from 'types';
+
+export type BasketItem = {
+  id: string;
+  title: string;
+  description: string;
+  price: number;
+  count: number;
+  imageUrl: string;
+};
+export interface IBasket {
+  isLoading: boolean;
+  items: any;
+  totalPrice: number;
+  totalItems: number;
+  userId: string;
+  basketObserver: Nullable<QuerySnapshot>;
+}
+
+const initialState: IBasket = {
+  isLoading: false,
+  items: null,
+  totalPrice: 0,
+  totalItems: 0,
+  userId: '',
+  basketObserver: null,
+};
+
+export const addBasketItemThunk = createAsyncThunk<void, BasketItem, ThunkApiConfig>(
+  'basketSlice/addBasketItemThunk',
+  async (values, { dispatch, rejectWithValue, getState }) => {
+    try {
+      const { id } = getState().user;
+
+      if (!id) return;
+
+      const data = {
+        id,
+        values,
+      };
+
+      await setDocSubCollectionById(
+        NamesDBCollection.Users,
+        NamesDBCollection.Basket,
```

Comment:
rename it to subcollection
```suggestion
        NamesDBSubcollection.Basket,
```